Dark Pit - nair rework, shine jc during orbitar release

krabsisa opened the pull request:
Guardian Orbitars can now be jump cancelled during the end part of the animation, similar to other shines. Previously you could only JC a held shine if the B button remained held. This should make the input a bit more comfortable especially on GameCube controller due to the distance between the B and XY buttons.

Reworked Dark Pit's nair to be a 2 hit move:
```
- Adjusted sound and effects to match the new properties
- Added a motion rate modifier to speed up the animation between the first and second hit
- FAF: 44
```

Hit 1:
```
- Active on: 4-9
- Damage: 6
- Angle: 65
- KBG: 45
- BKB: 50
```
![2022041617002600-0E7DF678130F4F0FA2C88AE72B47AFDF](https://user-images.githubusercontent.com/75001008/229640202-b3809ee2-d849-4340-8551-406d50ad4d2a.jpg)

Hit 2:
```
- Active on: 19
- Damage: 7
- Angle: 361
- KBG: 85
- BKB: 50
```
![2022041617003900-0E7DF678130F4F0FA2C88AE72B47AFDF](https://user-images.githubusercontent.com/75001008/229640209-48916b3d-c831-47be-a66f-005f649fe25e.jpg)

Videos (against no DI):


https://user-images.githubusercontent.com/75001008/229645423-62b28631-89df-46c4-bc06-b4262f1ee35f.mp4


https://user-images.githubusercontent.com/75001008/229645364-43318a58-d2d6-48de-aeef-36902b1ecf7d.mp4


https://user-images.githubusercontent.com/75001008/229645315-bb9ff7db-7511-4916-87aa-2768cc473c48.mp4

krabsisa committed:
Allow jump-cancelling DP shine during end of the move

krabsisa committed:
Dark pit nair rework

SuddyN commented on fighters/pitb/src/acmd/aerials.rs:
```diff
@@ -12,25 +65,23 @@ unsafe fn pitb_attack_air_n_game(fighter: &mut L2CAgentBase) {
-        ATTACK(fighter, 1, 0, Hash40::new("bowr"), 1.4, 363, 40, 0, 34, 4.0, 0.0, -6.5, 0.0, None, None, None, 0.75, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_F, false, 0, 0.0, 3, false, false, false, false, true, *COLLISION_SITUATION_MASK_A, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_cutup"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_CUTUP,  ...
-        ATTACK(fighter, 2, 0, Hash40::new("bowr"), 1.4, 80, 40, 0, 34, 4.0, 0.0, 6.5, 0.0, None, None, None, 0.75, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_F, false, 0, 0.0, 3, false, false, false, false, true, *COLLISION_SITUATION_MASK_G, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_cutup"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_CUTUP, *A ...
-        ATTACK(fighter, 3, 0, Hash40::new("bowr"), 1.4, 80, 40, 0, 34, 4.0, 0.0, -6.5, 0.0, None, None, None, 0.75, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_F, false, 0, 0.0, 3, false, false, false, false, true, *COLLISION_SITUATION_MASK_G, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_cutup"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_CUTUP, * ...
+        FT_MOTION_RATE(fighter, 0.700);
```
move this `FT_MOTION_RATE` call directly under the `frame` call above (outside the `is_excute` block)

SuddyN replied:
also, `FT_MOTION_RATE` calls should be in this format: `FT_MOTION_RATE(fighter, C / (B - A)` where:
A: current frame
B: frame you want to reach
C: number of frames to get from A to B

for example, if you want to get from frame 4 to frame 11 in 5 frames:
`FT_MOTION_RATE(fighter, 5.0 / (11.0 - 4.0)`

SuddyN commented on fighters/pitb/src/acmd/aerials.rs:
```diff
@@ -12,25 +65,23 @@ unsafe fn pitb_attack_air_n_game(fighter: &mut L2CAgentBase) {
     if is_excute(fighter) {
-        ATTACK(fighter, 0, 1, Hash40::new("top"), 4.0, 40, 85, 0, 70, 4.5, 0.0, 9.5, 3.0, None, None, None, 1.5, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_F, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_cutup"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_CUTUP, *AT ...
-        ATTACK(fighter, 1, 1, Hash40::new("top"), 4.0, 40, 85, 0, 70, 10.5, 0.0, 9.5, 3.0, None, None, None, 1.5, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_cutup"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_CUTUP,  ...
+        FT_MOTION_RATE(fighter, 1.000);
```
move this `FT_MOTION_RATE` call directly under the `frame` call above (outside the `is_excute` block)

SuddyN commented on fighters/pitb/src/acmd/aerials.rs:
```diff
@@ -12,25 +65,23 @@ unsafe fn pitb_attack_air_n_game(fighter: &mut L2CAgentBase) {
-        ATTACK(fighter, 0, 1, Hash40::new("top"), 4.0, 40, 85, 0, 70, 4.5, 0.0, 9.5, 3.0, None, None, None, 1.5, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_F, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_cutup"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_CUTUP, *AT ...
-        ATTACK(fighter, 1, 1, Hash40::new("top"), 4.0, 40, 85, 0, 70, 10.5, 0.0, 9.5, 3.0, None, None, None, 1.5, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_cutup"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_CUTUP,  ...
+        FT_MOTION_RATE(fighter, 1.000);
+        ATTACK(fighter, 1, 1, Hash40::new("top"), 7.0, 361, 85, 0, 50, 10.5, 1.0, 9.5, 3.0, None, None, None, 1.5, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_cutup"), *ATTACK_SOUND_LEVEL_L, *COLLISION_SOUND_ATTR_CUTUP, ...
```
considering the nature of this multihit, I suspect you will want this final hit to be centered in the same location as the first hitbox: `1.0, 8.5, 5.5`. If not, opponents at the edge of the hitbox will avoid the final hit

krabsisa replied:
It was intentionally placed a bit differently, but I see your point. I've made the 2nd hit a bit larger and moved its position to be closer to hit 1. Also changed the VFX to match this new position. Worth noting that the move won't always connect since it's more based on the user's drift and spacing, which is intended.

SuddyN commented on fighters/pitb/src/acmd/aerials.rs:
```diff
@@ -1,6 +1,59 @@
+    if is_excute(fighter) {
+        PLAY_STATUS(fighter, Hash40::new("se_pitb_attackair_n01"));
+    }
+    wait(lua_state, 14.0);
```
please use `frame` instead of `wait` here. You are targeting a specific frame with this call (smell: wait is longer that a couple frames)

krabsisa committed:
Further nair adjustments

SuddyN commented:
Tested this offline over the weekend, works well and feels great. New NAir has its advantages and disadvantages compared to old, I would describe it as a sidegrade.

hdr-balance-backroom reviewed (changes requested):
- Reduce the size of second hit of nair from 11.5 to 10.5
- Increase the active frame of second hit nair by 1 frame

krabsisa committed:
Merge remote-tracking branch 'upstream/dev' into darkpit-tweaks

krabsisa committed:
backroom requested changes

Reduce the size of second hit of nair from 11.5 to 10.5
Increase the active frame of second hit nair by 1 frame
